src/pool/srv_pool.c

```diff
@@ -2816,64 +2769,10 @@ ds_pool_attr_set_handler(crt_rpc_t *rpc)
 
 	ABT_rwlock_wrlock(svc->ps_lock);
 
-	rc = crt_bulk_get_len(in->pasi_bulk, &bulk_size);
-	if (rc != 0)
-		D_GOTO(out_lock, rc);
-	D_DEBUG(DF_DSMS, DF_UUID": count=%lu, size=%lu\n",
-		DP_UUID(in->pasi_op.pi_uuid), in->pasi_count, bulk_size);
-
-	D_ALLOC(data, bulk_size);
-	if (data == NULL)
-		D_GOTO(out_lock, rc = -DER_NOMEM);
-
-	sgl.sg_nr = 1;
-	sgl.sg_nr_out = sgl.sg_nr;
-	sgl.sg_iovs = &iov;
-	daos_iov_set(&iov, data, bulk_size);
-	rc = crt_bulk_create(rpc->cr_ctx, daos2crt_sg(&sgl),
-			     CRT_BULK_RW, &local_bulk);
-	if (rc != 0)
-		D_GOTO(out_mem, rc);
-
-	rc = attr_bulk_transfer(rpc, CRT_BULK_GET, local_bulk,
-				in->pasi_bulk, 0, 0, bulk_size);
-	if (rc != 0)
-		D_GOTO(out_bulk, rc);
-
-	names = data;
-	/* go to the end of names array */
-	for (values = names, i = 0; i < in->pasi_count; ++values)
-		if (*values == '\0')
-			++i;
-	sizes = (size_t *)values;
-	values = (char *)(sizes + in->pasi_count);
-
-	for (i = 0; i < in->pasi_count; i++) {
-		size_t len;
-		daos_iov_t key;
-		daos_iov_t value;
-
-		len = strlen(names) /* trailing '\0' */ + 1;
-		daos_iov_set(&key, names, len);
-		names += len;
-		daos_iov_set(&value, values, sizes[i]);
-		values += sizes[i];
-
-		rc = rdb_tx_update(&tx, &svc->ps_user, &key, &value);
-		if (rc != 0) {
-			D_ERROR(DF_UUID": failed to update attribute "
-				 "'%s': %d\n", DP_UUID(svc->ps_uuid),
-				 (char *) key.iov_buf, rc);
-			D_GOTO(out_bulk, rc);
-		}
-	}
-	rc = rdb_tx_commit(&tx);
-
-out_bulk:
-	crt_bulk_free(local_bulk);
-out_mem:
-	D_FREE(data);
-out_lock:
+	rc = ds_rsvc_set_attr(&svc->ps_rsvc, &tx, &svc->ps_user,
+			      in->pasi_bulk, rpc, in->pasi_count);
+	if (rc == 0)
+		rc = rdb_tx_commit(&tx);
```

Comment:
I think we prefer:

    rc = ds_rsvc_set_attr(...);
    if (rc != 0)
        goto out_lock;
    
    rc = rdb_tx_commit(&tx);

I've received similar review comments for my patches a couple of times in the past.

Reply:
Yeah.. I always get confused which one I should use. I think the goto is certainly more elegant and consistent with convention. I'll make the change.